src/fromager/commands/bootstrap.py

```diff
@@ -180,7 +180,15 @@ def write_constraints_file(
     conflicts = graph.get_install_dependency_versions()
     ret = True
     conflicting_deps = set()
-    for dep_name, nodes in sorted(conflicts.items()):
+
+    # Map for already resolved versions for a given dependency Eg: {"a": "0.4"}
+    resolved: dict[str, str] = {}
+
+    # List of unresolved dependencies
+    unresolved_dependencies = sorted(conflicts.items())
+
+    # Resolve dependencies with single version first. This will shrink the unresolved_dependencies to begin with.
+    for dep_name, nodes in unresolved_dependencies:
```

Comment:
```suggestion
    for dep_name, nodes in unresolved_dependencies[:]:
```

See comment on line 202